tests/integration/test_analysisd/test_syscollector/test_syscollector_events.py

```diff
@@ -0,0 +1,284 @@
+from wazuh_testing.tools.file import truncate_file
+from wazuh_testing.tools.services import control_service
+from wazuh_testing.vulnerability_detector import create_mocked_agent, delete_mocked_agent
+
+# Marks
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=0), pytest.mark.server]
+
+# Configurations
+TEST_DATA_PATH = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+messages_path = os.path.join(TEST_DATA_PATH, 'syscollector.yaml')
+with open(messages_path) as f:
+    test_cases = yaml.safe_load(f)
+
+
+# Fixtures
+@pytest.fixture(scope='module')
+def configure_custom_rules(get_configuration):
+    """Configure a syscollector custom rules for testing.
+    Restarting wazuh-analysisd is required to apply this changes.
+    """
+    source_rule = os.path.join(TEST_DATA_PATH, get_configuration['rule_file'])
+    target_rule = os.path.join(CUSTOM_RULES_PATH, get_configuration['rule_file'])
+
+    # copy custom rule with specific privileges
+    shutil.copy(source_rule, target_rule)
+    shutil.chown(target_rule, WAZUH_UNIX_USER, WAZUH_UNIX_GROUP)
+
+    yield
+
+    # remove custom rule
+    os.remove(target_rule)
+
+
+@pytest.fixture(scope='module')
+def restart_analysisd():
+    """wazuh-analysisd restart and log truncation"""
+    required_logtest_daemons = ['wazuh-analysisd']
+
+    truncate_file(ALERT_FILE_PATH)
+    truncate_file(LOG_FILE_PATH)
+
+    for daemon in required_logtest_daemons:
+        control_service('restart', daemon=daemon)
+
+    yield
+
+    for daemon in required_logtest_daemons:
+        control_service('stop', daemon=daemon)
+
+
+@pytest.fixture(scope='module')
+def mock_agent():
+    """Fixture to create a mocked agent in wazuh databases"""
+    control_service('stop', daemon='wazuh-db')
+    agent_id = create_mocked_agent(name="mocked_agent")
+    control_service('start', daemon='wazuh-db')
+
+    yield agent_id
+
+    control_service('stop', daemon='wazuh-db')
+    delete_mocked_agent(agent_id)
+    control_service('start', daemon='wazuh-db')
+
+
+@pytest.fixture(scope='module', params=test_cases, ids=[test_case['name'] for test_case in test_cases])
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+# Variables
+receiver_sockets_params = [(ANALYSISD_QUEUE_SOCKET_PATH, 'AF_UNIX', 'UDP')]
+receiver_sockets = None
+wazuh_log_monitor = FileMonitor(ALERT_FILE_PATH)
+alert_timeout = 5
+
+
+# Tests
+@pytest.mark.parametrize('test_case',
+                         list(test_cases),
+                         ids=[test_case['name'] for test_case in test_cases])
+def test_syscollector_events(test_case, get_configuration, mock_agent, configure_custom_rules, restart_analysisd,
+                             wait_for_analysisd_startup, connect_to_sockets_function):
+    '''
+    description:
+        Check if analysisd handle syscollector deltas properly by generating alerts.
+
+    wazuh_min_version:
+        4.3.0
```

Comment:
Is this correct or is it for 4.4?

Reply:
Yep! Target version was changed to 4.4.0